### ISSUE TITLE
test_NetworkCompare在img_conv上的单测未过。

### ISSUE THREAD

**Issue by luotao1** [PaddlePaddle/Paddle#1681]
合并 #1455 后，test_NetworkCompare在测试到[TEST(Compare, img_conv)](https://github.com/PaddlePaddle/Paddle/blob/develop/paddle/gserver/tests/test_NetworkCompare.cpp#L250)时出现下面问题（未合并前测过所有单测，是可以正常通过的）：
```
49: [       OK ] Compare.img_pool (47 ms)
49: [ RUN      ] Compare.img_conv
49: [INFO 2017-03-22 11:23:44,016 layers.py:2116] output for __conv_0__: c = 16, h = 16, w = 16, size = 4096
49: [INFO 2017-03-22 11:23:44,017 layers.py:2116] output for __conv_1__: c = 16, h = 16, w = 16, size = 4096
49: [INFO 2017-03-22 11:23:44,018 layers.py:2116] output for __conv_2__: c = 16, h = 16, w = 16, size = 4096
49: [WARNING 2017-03-22 11:23:44,019 networks.py:1444] `outputs` routine try to calculate network's inputs and outputs order. It might not work well.Please see follow log carefully.
49: [INFO 2017-03-22 11:23:44,019 networks.py:1472] The input order is [input]
49: [INFO 2017-03-22 11:23:44,019 networks.py:1478] The output order is [__concat_0__, __conv_2__]
49: [INFO 2017-03-22 11:23:44,023 layers.py:2116] output for __conv_0__: c = 16, h = 16, w = 16, size = 4096
49: [INFO 2017-03-22 11:23:44,024 layers.py:2116] output for __conv_1__: c = 16, h = 16, w = 16, size = 4096
49: [INFO 2017-03-22 11:23:44,026 layers.py:2116] output for __conv_2__: c = 16, h = 16, w = 16, size = 4096
49: [WARNING 2017-03-22 11:23:44,026 networks.py:1444] `outputs` routine try to calculate network's inputs and outputs order. It might not work well.Please see follow log carefully.
49: [INFO 2017-03-22 11:23:44,026 networks.py:1472] The input order is [input]
49: [INFO 2017-03-22 11:23:44,027 networks.py:1478] The output order is [__concat_0__, __conv_2__]
49: I0322 11:23:44.028722  2237 Trainer.cpp:165] trainer mode: Normal
49: I0322 11:23:44.029321  2237 GradientMachine.cpp:86] Initing parameters..
49: I0322 11:23:44.029422  2237 GradientMachine.cpp:93] Init parameters done.
49: I0322 11:23:44.056939  2237 test_NetworkCompare.cpp:207] forwardBackward of Network A is finished
49: I0322 11:23:44.056957  2237 Stat.cpp:130] ======= StatSet: [GlobalStatInfo] status ======
49: I0322 11:23:44.056964  2237 Stat.cpp:133] Stat=backward                       total=17.671     avg=0.883      max=0.916      min=0.867      count=20        
49: I0322 11:23:44.056980  2237 Stat.cpp:133] Stat=forward                        total=7.765      avg=0.388      max=0.407      min=0.383      count=20        
49: I0322 11:23:44.056990  2237 test_NetworkCompare.cpp:210] 
49: 
49: [WARNING 2017-03-22 11:23:44,060 networks.py:1444] `outputs` routine try to calculate network's inputs and outputs order. It might not work well.Please see follow log carefully.
49: [INFO 2017-03-22 11:23:44,060 networks.py:1472] The input order is [input]
49: [INFO 2017-03-22 11:23:44,060 networks.py:1478] The output order is [__concat_0__, __mixed_0__]
49: I0322 11:23:44.062014  2237 Trainer.cpp:165] trainer mode: Normal
49: F0322 11:23:44.062114  2237 hl_cuda_cudnn.cc:577] Check failed: CUDNN_STATUS_SUCCESS == cudnnStat (0 vs. 3) Cudnn Error: CUDNN_STATUS_BAD_PARAM
49: *** Check failure stack trace: ***
49:     @           0x94a0f0  google::LogMessage::Fail()
49:     @           0x94a04c  google::LogMessage::SendToLog()
49:     @           0x9499d0  google::LogMessage::Flush()
49:     @           0x94ca77  google::LogMessageFatal::~LogMessageFatal()
49:     @           0x917257  hl_create_filter_descriptor()
49:     @           0x6024a5  paddle::ConvBaseProjection::initCudnn()
49:     @           0x603066  paddle::ConvBaseProjection::ConvBaseProjection()
49:     @           0x6a9ef0  _ZNSt17_Function_handlerIFPN6paddle10ProjectionENS0_16ProjectionConfigESt10shared_ptrINS0_9ParameterEEbEZNS0_14ClassRegistrarIS1_IS3_S6_bEE13registerClassINS0_14ConvProjectionEEEvRKSsEUlS3_S6_bE_E9_M_invokeERKSt9_Any_dataS3_S6_b
49:     @           0x634fd4  paddle::Projection::create()
49:     @           0x6b0478  paddle::ConcatenateLayer2::init()
49:     @           0x6e245f  paddle::NeuralNetwork::init()
49:     @           0x6f3edb  paddle::GradientMachine::create()
49:     @           0x76bc88  paddle::TrainerInternal::init()
49:     @           0x7686ab  paddle::Trainer::init()
49:     @           0x5f6896  calcGradient()
49:     @           0x5f7b63  compareNetwork()
49:     @           0x5f84ba  Compare_img_conv_Test::TestBody()
49:     @           0xa5a11a  testing::internal::HandleSehExceptionsInMethodIfSupported<>()
49:     @           0xa5471e  testing::internal::HandleExceptionsInMethodIfSupported<>()
49:     @           0xa3a245  testing::Test::Run()
49:     @           0xa3aaec  testing::TestInfo::Run()
49:     @           0xa3b1b0  testing::TestCase::Run()
49:     @           0xa41b44  testing::internal::UnitTestImpl::RunAllTests()
49:     @           0xa5b218  testing::internal::HandleSehExceptionsInMethodIfSupported<>()
49:     @           0xa5550c  testing::internal::HandleExceptionsInMethodIfSupported<>()
49:     @           0xa40890  testing::UnitTest::Run()
49:     @           0x5de17b  main
49:     @       0x318ae1ecdd  (unknown)
49:     @           0x5f5805  (unknown)
49: /home/luotao/Paddle/paddle/.set_python_path.sh: line 42:  2237 Aborted                 (core dumped) $@
1/1 Test #49: test_NetworkCompare ..............***Failed    3.06 sec
```

初步怀疑是[`img_conv_b.conf`](https://github.com/PaddlePaddle/Paddle/blob/develop/paddle/gserver/tests/img_conv_b.conf)中`conv_projection`出错了。

**Comment by wangyang59** [comment]
@luotao1 Thanks for pointing this out. I will try to fix it as soon as possible.

**Comment by wangyang59** [comment]
@luotao1 I have fixed this bug and made a PR #1693 please have a look
